packages/react/src/components/Authenticator/ConfirmSignIn/ConfirmSignIn.tsx

```diff
@@ -20,14 +20,9 @@ export const ConfirmSignIn = ({
   const { isPending } = useAuthenticator((context) => [context.isPending]);
   const { handleChange, handleSubmit } = useFormHandlers();
 
-  const {
-    components: {
-      ConfirmSignIn: {
-        Header = ConfirmSignIn.Header,
-        Footer = ConfirmSignIn.Footer,
-      },
-    },
-  } = useCustomComponents();
+  const { ConfirmSignIn: components } = useCustomComponents();
+  const Header = components?.Header ?? ConfirmSignIn.Header;
+  const Footer = components?.Footer ?? ConfirmSignIn.Footer;
```

Comment:
Hopefully this is readable, but we are grabbing the `ConfirmSignIn` and assigning it to `components` then do some optional chaining and nullish coalescing. Reusing this pattern for other components.